### PR TITLE
Switch gravity repo CODEOWNERS to the newer terraformed teams

### PR DESCRIPTION
**Pull request by wadells**
## Description
This allows us to retire the codowers-gravity group and subgroups, which
are exceptions to the rule.

Currently @gravitational/codeowners-gravity-code is empty, meaning we can't easily merge to this repo.  Instead of fixing it again, I figured we should flip to use the sanctioned IT managed groups.

## Type of change
* Internal change

## Linked tickets and other PRs
N/A

## TODOs
- [ ] Address review feedback
- [ ] Ports
- [ ] (after merge of this and ports) Nuke the old gravity-codeowners groups

## Testing done
None.

## Additional Information
I'll backport this to all active branches after we've agreed the updates look correct.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (wadells) [inline .github/CODEOWNERS:8]: I want something like @gravitational/admins here, but no gravity team members are currently part of that group.

**Review by adaadb6 — APPROVED** [review]

**Review by knisbet — APPROVED** [review]

**Comment by wadells** [comment]
@knisbet can you merge this?  I've got a deadlock scenario where I can't merge #2606 because there aren't any codowners due to this, but I can't merge this because of the robotest error #2606 addresses.

Also, upon second though I'm not entirely sure this is going to route approvals correctly.  @gravitational/gravity-team has 7 members (including @NajiObeid @webvictim @dboslee and @ulysseskan) -- wheras this was previously limited to @knisbet, @bernardjkim, and @wadells.  Notably, I don't want to spam the previous 4 with review requests that aren't relevant.

**Comment by wadells** [comment]
Closing for now, as @adaadb6 repopulated @gravitational/codeowners-gravity-code

**Pull request closed without merging**
